File: cpp/creational/prototype/nishith/Prototype/Prototype/BotPlayer.cpp
#include "BotPlayer.h"
#include <iostream>

BotPlayer::BotPlayer(const std::string& name, int health, int damage, int bot_level)
   : Player(name, health, damage), bot_level_(bot_level)
{

}

Player* BotPlayer::clone() const
{
   return new BotPlayer(*this);
}

void BotPlayer::print() const
{
   std::cout << "BotPlayer: " << name_ << ", Health: " << health_
      << ", Damage Output: " << damage_
      << ", Bot Level: " << bot_level_ << std::endl;
}

File: java/creational/prototype/praful/BotPlayer.java
public class BotPlayer extends Player {
    private String bomb;

    public BotPlayer(String name, String bomb, int health, int level) {
        super(name, health, level); // Call the constructor of Player with the name, health, and level
        this.bomb = bomb; // Initialize the bomb type
    }

    public BotPlayer(BotPlayer botPlayer) {
        super(botPlayer.getName(), botPlayer.getHealth(), botPlayer.getLevel()); // Copy constructor
        this.bomb = botPlayer.bomb; // Copy the bomb type
    }

    @Override
    public Player clone() {
        return new BotPlayer(this); // Create a new BotPlayer using the copy constructor
    }

    @Override
    public void print() {
        System.out.println("BotPlayer Name: " + getName());
        System.out.println("Health: " + getHealth());
        System.out.println("Level: " + getLevel());
        System.out.println("Bomb: " + bomb);
    }
}
